docs/mkdocs/docs/api/macros/json_diagnostic_positions.md

```diff
@@ -10,9 +10,12 @@ When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nl
 that JSON object/field in the original string the object was parsed from. Likewise, `end_pos()` returns the end position of that JSON object/field in the
 original string the object was parsed from.
 
-For objects and arrays, the start and end positions represent the positions of the opening and closing braces or brackets, respectively. For fields, the start
-and end positions represent either the opening and closing quotes for that field's value or the first and character after last in the field's numerical or
-predefined true/false/null values.
+For objects and arrays, the start position represents the position of the opening brace or bracket. The end position is the character after the closing
```

Comment:
I will leave it up to you, but I prefer a version of separating talking about what start/end means vs what the first and last character is for each type.